services/dedup/dedup.go

```diff
@@ -40,24 +29,38 @@ func DefaultPath() string {
 }
 
 // New creates a new deduplication service. The service needs to be closed after use.
-func New(path string, fns ...OptFn) Dedup {
+func New(path string) Dedup {
 	var dedupWindow time.Duration
 	config.RegisterDurationConfigVariable(3600, &dedupWindow, true, time.Second, []string{"Dedup.dedupWindow", "Dedup.dedupWindowInS"}...)
 	log := logger.NewLogger().Child("dedup")
+	defer func() {
+		// TODO : Remove this after badgerdb v2 is completely removed
+		tmpDirPath, _ := misc.CreateTMPDIR()
+		_ = os.RemoveAll(fmt.Sprintf(`%v%v`, tmpDirPath, "/badgerdbv2"))
+	}()
+	badgerOpts := badger.
+		DefaultOptions(path).
+		WithCompression(options.None).
+		WithIndexCacheSize(16 << 20). // 16mb
+		WithNumGoroutines(1).
+		WithNumMemtables(config.GetInt("BadgerDB.numMemtable", 1)).
+		WithValueThreshold(config.GetInt64("BadgerDB.valueThreshold", 1048576)).
+		WithBlockCacheSize(0).
+		WithNumVersionsToKeep(1).
+		WithNumLevelZeroTables(config.GetInt("BadgerDB.numLevelZeroTables", 5)).
+		WithNumLevelZeroTablesStall(config.GetInt("BadgerDB.numLevelZeroTablesStall", 15)).
+		WithSyncWrites(config.GetBool("BadgerDB.syncWrites", false))
```

Comment:
we can add the following, since we are not using transactions where we reading before we are writing
```go
WithDetectConflicts(false)
```